src/commonMain/kotlin/fr/acinq/lightning/payment/OfferManager.kt

```diff
@@ -187,10 +187,10 @@ class OfferManager(val nodeParams: NodeParams, val walletParams: WalletParams, v
         val needIntermediateHop = when (destination) {
             is Destination.BlindedPath -> when (val introduction = destination.route.introductionNodeId) {
                 is EncodedNodeId.Plain -> introduction.publicKey != remoteNodeId
-                is EncodedNodeId.ShortChannelIdDir -> true // we don't have access to the graph data and rely on our peer to resolve the scid
+                else -> true // we don't have access to the graph data and rely on our peer to resolve the scid
```

Comment:
Can you explicitly handle every case here instead of using `else`? This ensures that if we add another `EncodedNodeId` type in the future, the compiler will force us to think about what value to use here.

Reply:
Done